deriva/core/ermrest_model.py

```diff
@@ -1716,14 +1718,25 @@ def alter(
             self.schema.tables[self.name] = self
 
         if 'schema_name' in changes:
+            old_schema_name = self.schema.name
             del self.schema.tables[self.name]
             self.schema = self.schema.model.schemas[changed['schema_name']]
+            for key in self.keys:
+                if key.constraint_schema:
+                    key.constraint_schema = self.schema
```

Comment:
@karlcz I added this to (I believe) fix a bug in that the existing method wasn't updating the `key.constraint_schema`. Unless I misunderstood something... The behavior I was getting before adding this code is that that `model.apply()` was throwing a `KeyError` exception.

Reply:
Good catch